FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90

```diff
@@ -5677,7 +5677,38 @@ subroutine moist_bud2(im,ix,ix2,levs,me,kdt,grav,dtp,delp,rain, &
 
       end subroutine moist_bud2
 
-      subroutine moist_cv_nwat6(initial_dynamics_q, physics_q, pressure_on_interfaces, &
+      subroutine compute_q_liquid(im, levs, nwat, ntcw, ntrw, new_dynamics_q, q_liquid)
+        integer, intent(in) :: im, levs, nwat, ntcw, ntrw
+        real(kind=kind_phys), dimension(1:im,1:levs,1:nwat), intent(in) :: new_dynamics_q
+        real(kind=kind_phys), dimension(1:im,1:levs), intent(out) :: q_liquid
+
+        q_liquid = 0.0
+        if (ntcw .gt. 0) then
+          q_liquid = q_liquid + new_dynamics_q(:,:,ntcw)
+        endif
+        if (ntrw .gt. 0) then
+          q_liquid = q_liquid + new_dynamics_q(:,:,ntrw)
+        endif
+      end subroutine compute_q_liquid
+
+      subroutine compute_q_ice(im, levs, nwat, ntiw, ntsw, ntgl, new_dynamics_q, q_ice)
+        integer, intent(in) :: im, levs, nwat, ntiw, ntsw, ntgl
+        real(kind=kind_phys), dimension(1:im,1:levs,1:nwat), intent(in) :: new_dynamics_q
+        real(kind=kind_phys), dimension(1:im,1:levs), intent(out) :: q_ice
+
+        q_ice = 0.0
+        if (ntiw .gt. 0) then
+          q_ice = q_ice + new_dynamics_q(:,:,ntiw)
+        endif
+        if (ntsw .gt. 0) then
+          q_ice = q_ice + new_dynamics_q(:,:,ntsw)
+        endif
+        if (ntgl .gt. 0) then
+          q_ice = q_ice + new_dynamics_q(:,:,ntgl)
+        endif
+      end subroutine compute_q_ice
+
+      subroutine moist_cv(initial_dynamics_q, physics_q, pressure_on_interfaces, &
```

Comment:
So this is basically the same as `moist_cv` in `fv_mapz` except it doesn't support the special case and expect data dimension of `(i*j, 1, k)` instead of `(i,j,k)`. I think we should avoid using identical subroutine name to avoid confusion, maybe change it to `moist_cv_gfs` or `moist_cv_physics`?

Reply:
I'm personally a fan of overloading names if they are in different modules. Isn't the suffix  "_physics" in your suggestion implied by the name of this module which is "module_physics_driver"?  The name `module_physics_driver.moist_cv_physics` seems a bit redundant.

Reply:
Fair enough. Ideally there will just be one subroutine that works for both, but that's probably outside of this PR. We are planning on using the same `moist_cv` and `moist_cp` in the python port.

Reply:
I totally agree it would be nice to be able to use the same subroutine for both (or at least share more of the code).  It would indeed require a fair amount of refactoring to do that in the fortran -- as you noted the different array shapes, different data types, and even the build system make it awkward at the moment.  That's great that you will be able to do that in the Python port.

@elynnwu are you ok if I go ahead and merge?